src/lib/network_pool/transaction_pool.ml

```diff
@@ -1249,7 +1255,7 @@ struct
                                         ~is_sender_local uc e
                                     with
                                     | `Reject ->
-                                        failure := Error (Command_failure e) ;
+                                        add_failure (Command_failure e) ;
```

Comment:
I’m not sure if this change makes sense to me. The outer loop is for senders and the inner loop is for the commands from a sender. If one of the commands from a sender errors, we should discard the rest and try to add commands from other senders(which we do). So maybe we just need to reset `failure` to `Ok ()` for each sender?

Reply:
The existing behavior was to return the last error seen, discarding previously-seen errors. Don't we want to report all errors?

The proposed change doesn't change the control flow at all, it just reports more errors. We still add commands from as many senders as possible.

Reply:
How do you mean? It a doing a map so it should return the results per sender? further down we filter this list and use only the ones that succeeded

Reply:
Oh I see what you mean

Reply:
Looks like we are discarding the diff even if there is a single error unless `allow_failures_for_tests` is true

Reply:
This makes sense since we don't want to spend time processing someone's diff that contains an invalid txn . But we should definitely optimize that for when `allow_failures_for_tests` is false by returning from the outer loop as soon as there is an error in the inner loop. I'm creating another issue for this